src/main/scala/com/lightbend/kafkalagexporter/PrometheusEndpointSink.scala

```diff
@@ -12,34 +12,41 @@ import io.prometheus.client.{CollectorRegistry, Gauge}
 import scala.util.Try
 
 object PrometheusEndpointSink {
-  def apply(httpPort: Int, definitions: MetricDefinitions): MetricsSink =
-    Try(new PrometheusEndpointSink(httpPort, definitions))
+  def apply(appConfig: AppConfig, definitions: MetricDefinitions): MetricsSink =
+    Try(new PrometheusEndpointSink(appConfig, definitions))
       .fold(t => throw new Exception("Could not create Prometheus Endpoint", t), sink => sink)
 }
 
-class PrometheusEndpointSink private(httpPort: Int, definitions: MetricDefinitions) extends MetricsSink {
+class PrometheusEndpointSink private(appConfig: AppConfig, definitions: MetricDefinitions) extends MetricsSink {
 
-  private val server = new HTTPServer(httpPort)
+  private val server = new HTTPServer(appConfig.port)
   private val registry = CollectorRegistry.defaultRegistry
 
   DefaultExports.initialize()
 
-  private val metrics: Map[GaugeDefinition, Gauge] = definitions.map(definition =>
-    definition -> Gauge.build()
-      .name(definition.name)
-      .help(definition.help)
-      .labelNames(definition.labels: _*)
-      .register(registry)
-  ).toMap
+  private val metrics: Map[String, Map[GaugeDefinition, Gauge]] = {
+    appConfig.clusters.map { cluster =>
+      val globalLabelNamesForCluster = appConfig.globalLabelsForCluster(cluster.name).map(_._1).toSeq
+      cluster.name -> definitions.map(definition =>
+        definition -> Gauge.build()
+          .name(definition.name)
+          .help(definition.help)
+          .labelNames(globalLabelNamesForCluster ++ definition.labels: _*)
+          .register(registry)
+      ).toMap
+    }.toMap
+  }
 
   override def report(m: MetricValue): Unit = {
-    val metric = metrics.getOrElse(m.definition, throw new IllegalArgumentException(s"No metric with definition ${m.definition.name} registered"))
-    metric.labels(m.labels: _*).set(m.value)
+    val metric = getMetricsForClusterName(m.definition, m.clusterName)
+    val globalLabelValuesForCluster = appConfig.globalLabelsForCluster(m.clusterName).map(_._2).toSeq
+    metric.labels(globalLabelValuesForCluster ++ m.labels: _*).set(m.value)
   }
 
 
-  override def remove(m: RemoveMetric): Unit =
-    metrics.get(m.definition).foreach(_.remove(m.labels: _*))
+  override def remove(m: RemoveMetric): Unit = {
+    metrics.foreach(_._2.foreach(_._2.remove(m.labels: _*)))
```

Comment:
Please match the tuple value.  It's ok to unravel this one-liner for readability.